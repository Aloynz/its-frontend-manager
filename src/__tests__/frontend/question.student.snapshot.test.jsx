import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import QuestionView from '@/app/courses/[courseId]/[questionId]/page'

// mock useRouter
jest.mock("next/navigation", () => ({
    useRouter() {
      return {
        prefetch: () => null
      };
    }
  }));

// mock courseId param
const mockQuestionParams = {
    questionId: 'mockQuestionId',
    courseId: 'mock123'
}
  
// mock server action getUserProps
jest.mock('@/actions/getUserProps', () => {

    // mock student submissions
    const mockSubmissions = [{
        id: 'mockSubmissionId',
        questionId: 'mockQuestionId',
        user_id: 'mockId1',
        submitted_program: 'mockProgram',
        feedback: '',
        grade: 0,
    }]

    return {
        getUserProps: jest.fn().mockResolvedValue({
            id: 'mockId1',
            email: 'mockEmail1',
            password: 'mockPw1',
            role: 'STUDENT',
            school_id: 'mockSchoolId',
            submissions: mockSubmissions,
        }),
    }
    
})

// mock server action getQuestionInfo
jest.mock('@/actions/getQuestionInfo', () => {

    return {
        getQuestionInfo: jest.fn().mockResolvedValue({
            id: 'mockQuestionId',
            title: 'mockQuestion',
            description: 'mockDescription',
            reference_program: '',
            courseId: 'mockSchoolId_mock123',
            course: {
                code: 'mock123'
            }
        })
    }    
})

it("renders student's question page unchanged", async () => {
    const result = await QuestionView({ params: mockQuestionParams })
    const { container } = render(result);
    expect(container).toMatchSnapshot();
  });