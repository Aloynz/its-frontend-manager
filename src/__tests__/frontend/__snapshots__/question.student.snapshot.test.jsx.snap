// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders student's question page unchanged 1`] = `
<div>
  <div
    class="flex h-full w-full data-[panel-group-direction=vertical]:flex-col min-h-[calc(100vh-100px)]"
    data-panel-group=""
    data-panel-group-direction="horizontal"
    data-panel-group-id=":r0:"
    style="display: flex; flex-direction: row; height: 100%; overflow: hidden; width: 100%;"
  >
    <div
      class=""
      data-panel=""
      data-panel-group-id=":r0:"
      data-panel-id=":r1:"
      data-panel-size="30.0"
      style="flex-basis: 0px; flex-grow: 30; flex-shrink: 1; overflow: hidden;"
    >
      <div
        class="pr-4 h-full"
      >
        <div
          class="h-5/6 flex flex-col space-y-2"
        >
          <h1
            class="text-2xl font-semibold"
          >
            mockQuestion
          </h1>
          <p>
            mockDescription
          </p>
        </div>
        <div
          class="h-1/6 flex items-end pb-4"
        >
          <a
            href="/courses/mock123"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            >
              Return to course
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      aria-controls=":r1:"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="30"
      class="relative flex w-px items-center justify-center bg-border after:absolute after:inset-y-0 after:left-1/2 after:w-1 after:-translate-x-1/2 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 data-[panel-group-direction=vertical]:h-px data-[panel-group-direction=vertical]:w-full data-[panel-group-direction=vertical]:after:left-0 data-[panel-group-direction=vertical]:after:h-1 data-[panel-group-direction=vertical]:after:w-full data-[panel-group-direction=vertical]:after:-translate-y-1/2 data-[panel-group-direction=vertical]:after:translate-x-0 [&[data-panel-group-direction=vertical]>div]:rotate-90"
      data-panel-group-direction="horizontal"
      data-panel-group-id=":r0:"
      data-panel-resize-handle-enabled="true"
      data-panel-resize-handle-id=":r2:"
      data-resize-handle=""
      data-resize-handle-state="inactive"
      role="separator"
      style="user-select: none;"
      tabindex="0"
    >
      <div
        class="z-10 flex h-4 w-3 items-center justify-center rounded-sm border bg-border"
      >
        <svg
          class="lucide lucide-grip-vertical h-2.5 w-2.5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="9"
            cy="12"
            r="1"
          />
          <circle
            cx="9"
            cy="5"
            r="1"
          />
          <circle
            cx="9"
            cy="19"
            r="1"
          />
          <circle
            cx="15"
            cy="12"
            r="1"
          />
          <circle
            cx="15"
            cy="5"
            r="1"
          />
          <circle
            cx="15"
            cy="19"
            r="1"
          />
        </svg>
      </div>
    </div>
    <div
      class=""
      data-panel=""
      data-panel-group-id=":r0:"
      data-panel-id=":r3:"
      data-panel-size="70.0"
      style="flex-basis: 0px; flex-grow: 70; flex-shrink: 1; overflow: hidden;"
    >
      <div
        class="h-full"
      >
        <div
          class="h-4/6"
        >
          <section
            style="display: flex; position: relative; text-align: initial; width: 100%; height: 100%;"
          >
            <div
              style="display: flex; height: 100%; width: 100%; justify-content: center; align-items: center;"
            >
              Loading...
            </div>
            <div
              style="width: 100%; display: none;"
            />
          </section>
        </div>
        <div
          class="shrink-0 bg-border h-[1px] w-full"
          data-orientation="horizontal"
          role="none"
        />
        <div
          class="h-1/6"
        >
          <div
            class="border ml-4 h-full"
          >
            <p>
              No ITS Feedback
            </p>
          </div>
        </div>
        <div
          class="h-1/6 flex items-end justify-end pb-4"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mr-2"
          >
            Run Check
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
          >
            Submit Code
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
